File: api/Channel/src/main/java/com/vibio/channel/model/Channel.java
/*
 *  Channel
 *  @author: Minhhieuano
 *  @created 9/27/2024 10:25 PM
 * */

package com.vibio.channel.model;

import com.aventrix.jnanoid.jnanoid.NanoIdUtils;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Channel {

	@Id
	@Builder.Default
	private String id = NanoIdUtils.randomNanoId();

	@Column(unique = true)
	private String accountId;

	@Column(unique = true)
	private String name;

	private String thumbnail;

	private String defaultEmail;

	private String background;

	private String description;

	@CreationTimestamp
	private LocalDateTime createdAt;

	@UpdateTimestamp
	private LocalDateTime updatedAt;
}

File: vibio_ui/src/common/type/channel.ts
export type ChannelBasicInfo = {
    id: string;
    name: string;
    thumbnail: string;
};

export type Channel = {
    id: string;
    name: string;
    accountId: string;
    background: string;
    thumbnail: string;
    member: boolean;
    subscribeCount: number;
    subscribed: boolean;
    description: string;
};

export type ChannelDetail = {
    id: string;
    name: string;
    description: string;
    thumbnail: string;
    background: string;
    defaultEmail: string;
    createdAt: string;
    updatedAt: string;
};
